### ISSUE TITLE
FixedLenByteArray support

### ISSUE THREAD

**Issue by kylebarron** [mcaceresb/stata-parquet#7]
> Reading FixedLenByteArray and Int96 variables are not supported, as they have no direct Stata counterpart

`FixedLenByteArray` is actually closer to a direct Stata counterpart. It's my understanding that with `ByteArray`, you have no idea how many bytes are in each row of the column, so you have to set `str255`, import the data, and then compress. 

`FixedLenByteArray` is set to have the same width of bytes in every row.

I'm not sure how prevalent this field is in real-world Parquet data. Unless you have a column where every row is the same length, like an MD5 hash, the variable length encoding is much more space-efficient.

In any case, reading in all string columns as `str255` is kind of untenable for large datasets with lots of strings. I'm not sure what the best way to get around this is... Set each `str#` value low, and then make Stata recast the type to a larger string as you're importing the data?

**Comment by mcaceresb** [comment]
> FixedLenByteArray is actually closer to a direct Stata counterpart.

I'm not terribly sure that's true, but you might be right. (Because while Stata does allocate a str# as fixed width, each individual string can be of varying width; I think this uses null-terminations and I'm not entirely sure how it maps to a FLBA exactly). In any case, there is some support for this in the latest  commit.

>  reading in all string columns as str255 is kind of untenable for large datasets with lots of strings. 

Indeed. For ByteArray data I don't really know how to get around this either. Hopefully there turns out to be a way to extract the largest array length from it.

**Comment by kylebarron** [comment]
Here's an open PR for `FixedLenByteArray` in Spark. https://github.com/apache/spark/pull/20826

So `ByteArray` is way more popular in the wild.

**Comment by mcaceresb** [comment]
Dan made a good point noting that we would expect FixedLenByteArray to be better at storing string IDs (e.g. encrypted bene id, hashes, etc.). Even if stored as binary instead of strings they would still be fixed width.

For arbitrary strings, which I assume are much more common than encrypted IDs and hashes, ByteArray would be the way to go because it saves space.

**Comment by kylebarron** [comment]
> we would expect FixedLenByteArray to be better at storing string IDs 

Yes this is true, with `FixedLenByteArray` they might all be 10 bytes. With `ByteArray` they might all be 11 bytes, with the extra byte saying that each string is 10 bytes long.

It's fine to implement `FixedLenByteArray`, but until they're supported in Spark, it's not realistic to use data in that format in production.
